opm/input/eclipse/Schedule/Well/WellTestConfig.hpp

```diff
@@ -60,6 +60,7 @@ enum class Reason {
     GROUP = 4,
     THP_DESIGN=8,
     COMPLETION=16,
+    CONVERGENCE=32,
```

Comment:
I suppose this is fine, but you need to be aware that we'll [throw an exception](https://github.com/OPM/opm-common/blob/034b510205f219335dd01f8b7770990102df6b12/opm/input/eclipse/Schedule/Well/WellTestState.cpp#L55-L56), and not create any restart file output. if the well is closed due to `CONVERGENCE` at restart file writing time.

Reply:
That's not good. Can we return WTest::EclCloseReason::NONE for CONVERGENCE to avoid adding new reasons to the restart files? Otherwise, we may break restart compatibility. The force-shut option for non-convergent wells is a back-door solution when we cannot converge the well and properly shut it down due to operability. Currently, they are tagged as PHYSICAL. But since the purpose of these PRs is to allow for checking these wells regardless of WTEST I think setting them as NONE is more robust.

Reply:
Do we ever shut wells due to `case WellTestConfig::Reason::NONE`?

Reply:
I think so. It is better to open this well than to potentially shut it forever (like we do now). It will break perfect restart, but I think that is better than breaking restart compatibility.